### ISSUE TITLE
Resolve buzzer shield requiring reset

### ISSUE THREAD

**Issue by HowManyOliversAreThere** [mattytrentini/micropython_workshop#24]
On the ESP32 board, the buzzer shield will only make noise once before having to be reset to do it again. The cause of this should be identified to ease the user experience when developing / testing.

**Comment by HowManyOliversAreThere** [comment]
Re-creating the `buzzer_pin` object (`buzzer_pin = machine.Pin(d1_mini.D6, machine.Pin.OUT)`) is what causes the buzzer to cease functioning. This suggests that there is an issue in MicroPython ESP32 port with the PWM channel potentially not getting correctly deinitialised when the pin object is torn down.

This can be worked around by explicitly calling `buzzer.deinit()` when you are finished using the buzzer, and then re-initialising the buzzer via `buzzer = machine.PWM(buzzer_pin)` when you wish to use it again.

Proposal for resolving this issue in regards to workshop docs is just to provide this deinit / reinit info, and separately raising this as an issue on the official MicroPython repo - @mattytrentini does that seem reasonable to you?

**Comment by mattytrentini** [comment]
Yes, agree - though we really will need to update the documentation when a fix is available. It's not ideal to force beginner users to use deinit - shouldn't be necessary! 

Description looks good, thanks!

**Comment by HowManyOliversAreThere** [comment]
@mattytrentini Do you agree that this issue should be closed given that a solution has been put in place? If a futher update is expected once MicroPython itself is fixed to make this work as expected (without `deinit()`), then I believe this should be captured in a separate issue.
